### PR TITLE
Migrating bot commands

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
# PR Title
### Checklist

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines](/CONTRIBUTING.md).
- [x] I agree with the [license](/LICENSE).

### Changes
- [ ] Refactoring
- [x] Internal code
- [ ] External Code / Libraries
- [ ] Documentation
- [ ] Other: \_____

Closes Issue: -

## Description
I have migrated the bot commands to slash commands.

### CODE REVIEW

**Comment by RealYusufIsmail** [comment]
Help command left

File: src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java [hunk 1]
```diff
@@ -1,47 +1,49 @@
...
+
+		if (status.equalsIgnoreCase("replace")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.REPLACE);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("remove"))), false).build()).queue();
+		} else if (status.equalsIgnoreCase("delete")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.DELETE);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("delete"))), false).build()).queue();
+		} else if (status.equalsIgnoreCase("off")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.OFF);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("off"))), false).build()).queue();
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java:44]: Yet another occurance of shitty legacy code, please do it like [this](https://github.com/Black0nion/BlackOnion-Bot/blob/main/src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSwearCommand.java#L32-L43).

Also, thanks for replacing the Utils.checkPermission

File: src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java [hunk 2]
```diff
@@ -1,47 +1,49 @@
...
 		} else {
-			message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField(LanguageSystem.getTranslation("antispoilerstatus", author, guild).replace("%status%", LanguageSystem.getTranslation(guild.getAntiSpoilerType().name(), author, guild)), LanguageSystem.getTranslation("howtoantispoilertoggle", author, guild).replace("%command%", "``" + CommandEvent.getCommandHelp(guild, this) + "``"), false).build()).queue();
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java:44]: I'd still keep this, but with the new way of responding.
This way, you can still check the status of the system

File: src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java [hunk 3]
```diff
@@ -1,47 +1,49 @@
...
 
 	public AntiSpoilerCommand() {
-		this.setCommand("antispoiler", "as")
-			.setSyntax("[replace | delete | off]")
-			.setRequiredPermissions(Permission.MESSAGE_MANAGE);
+		super(builder(Commands.slash("antispoiler", "Used to deleted/disable/replace the anti-spoiler system.")
+				.addOptions(new OptionData(OptionType.STRING, STATUS, "The status", true)
+						.addChoice("replace", "replace")
+						.addChoice("delete", "delete")
+						.addChoice("off", "off")))
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java:28]: Please make this optional to allow checking the status of the system (see below)

File: src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSwearCommand.java [hunk 4]
```diff
@@ -1,45 +1,50 @@
...
+		if(status.equals(REPLACE)) {
+			guild.setAntiSwearType(AntiSwearType.REPLACE);
+			cmde.success("antiswearstatuschanged", "antiswearis", new Placeholder("status", cmde.getTranslation(REPLACE)));
+		} else if(status.equals(OFF)) {
+			guild.setAntiSwearType(AntiSwearType.OFF);
+			cmde.success("antiswearstatuschanged", "antiswearis", new Placeholder("status", cmde.getTranslation(OFF)));
+		} else if(status.equals(DELETE)) {
+			guild.setAntiSwearType(AntiSwearType.DELETE);
+			cmde.success("antiswearstatuschanged", "antiswearis", new Placeholder("status", cmde.getTranslation(DELETE)));
 		} else {
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSwearCommand.java:46]: Why did you replace the "generic" variant with hardcoded ifs?

File: src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSwearCommand.java [hunk 5]
```diff
@@ -1,45 +1,50 @@
...
-			message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField(LanguageSystem.getTranslation("antiswearstatus", author, guild).replace("%status%", LanguageSystem.getTranslation(guild.getAntiSwearType().name(), author, guild)), LanguageSystem.getTranslation("howtoantisweartoggle", author, guild).replace("%command%", "``" + CommandEvent.getCommandHelp(guild, this) + "``"), false).build()).queue();
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSwearCommand.java:42]: Same as AntiSpoilerCommand, please still allow checking the status

File: src/main/java/com/github/black0nion/blackonionbot/commands/bot/HelpCommand.java [hunk 6]
```diff
@@ -1,112 +1,109 @@
...
-				// a command
-				for (final Map.Entry<String[], TextCommand> entry : CommandBase.commandsArray.entrySet()) {
-					final TextCommand cmd = entry.getValue();
-					if (cmd.isVisible(author) && Arrays.asList(entry.getKey()).contains(args[1])) {
-						cmde.success("help", CommandEvent.getCommandHelp(guild, cmd), cmde.getTranslationOrEmpty("help" + cmd.getCommand()[0].toLowerCase()));
-						return;
-					}
+	public void execute(@NotNull SlashCommandEvent cmde, @NotNull SlashCommandInteractionEvent e, @NotNull BlackMember member, @NotNull BlackUser author, BlackGuild guild, @NotNull TextChannel channel) {
+		var command = e.getOption(COMMAND_NAME, OptionMapping::getAsString);
+		if(command != null) {
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/bot/HelpCommand.java:47]: Missing space

File: src/main/java/com/github/black0nion/blackonionbot/commands/bot/SupportCommand.java [hunk 7]
```diff
@@ -1,32 +1,33 @@
...
 import java.util.stream.Collectors;
 
 import static com.github.black0nion.blackonionbot.utils.config.Config.metadata;
 
-public class SupportCommand extends TextCommand {
+public class SupportCommand extends SlashCommand {
 
 	public SupportCommand() {
-		this.setCommand("support", "authors");
+		super(builder(Commands.slash("support", "Used to get support.")));
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/bot/SupportCommand.java:20]: If you want, you can use `super("support", "Used to get support.")` directly, but this is also completely fine

**Review by SIMULATAN — CHANGES_REQUESTED** [review]
Some nitpicking, sorry for that 😬

Also, I haven't tested it yet, I'll do that in like 30 mins or something

**Comment by RealYusufIsmail** [comment]
Seems like everything works

**Comment by SIMULATAN** [comment]
TODO for me: recode /help command with paginator

**Review by SIMULATAN — APPROVED** [review]

**Pull request merged**